### PR TITLE
Adding smoke test target and specific tests for build on PR

### LINKED ISSUE #643: systest/Makefile changes need to be performed for smoke tests to incorporate Jenkins PR runs
* Title: systest/Makefile changes need to be performed for smoke tests to incorporate Jenkins PR runs
* Attachments: None
* Details: This is an enhancement for internal Jenkins test runs automated upon PR submitting

#### OpenStack Release
Liberty; however, it should also be reflected in Mitaka and Newton and any subsequent releases.

#### Description
This enhancement should have a means by which to globally inform the subsequently-run scripts that this is a 'smoke' run that does not need to log all verbosity.  This should also be easily switched off.

### PR DESCRIPTION
**Pull request by pjbreaux**
@ssorenso 
#### What issues does this address?
Fixes #643 

#### What's this change do?
* This introduces a 'run_smoke_tests' target
* Leaves the majority of run_tests target unmodified

#### Any background context?
We are currently not running functional tests on PRs. We need to get a
set of core tests that can be run on PR and when they pass, we can merge.
Note that this is in addition to the manual tests that developers will perform.

Ran tests in the test instance of Jenkins I have running in my openstack
environment. Have only been using it against my fork, but will now
change that to main fork.

### CODE REVIEW

**Comment by pjbreaux** [comment]
test this please

**Comment by pjbreaux** [comment]
test this please

**Comment by zancas** [comment]
@pjbreaux we merged @ssorenso 's 13.0.0 image update into liberty you should fetch-and-rebase.

**Comment by pjbreaux** [comment]
retest this please

**Comment by pjbreaux** [comment]
@ssorenso: Ok, the the build on PR is working correctly. Due to [this issue](https://github.com/F5Networks/neutron-lbaas/issues/42), we are seeing intermittent failures in scenario tests. At least we are getting tests run and are able to see the passes and failures.

File: systest/Makefile [hunk 1]
```diff
@@ -128,6 +128,10 @@ run_agent_transfers:
 	@echo "running tests extracted from 'f5-openstack-agent' project..."
 	cd scripts && ./disallow_dangerous_vars.sh && ./$@.sh
 
+run_smoke_tests:
+	@echo "running smoke tests for build on PR..."
+	cd scripts && ./disallow_dangerous_vars.sh && ./$@.sh
```
Comment (zancas) [inline systest/Makefile]: disallow dangerous variables goes away in `mitaka` it looks like there won't be a conflict, so this is just an FYI.

File: systest/scripts/run_smoke_tests.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Create .pytest.rootdir file at root of the neutron-lbaas repository directory
+touch ${NEUTRON_LBAAS_DIR}/.pytest.rootdir
+
+# LBaaSv2 Scenario test cases with F5 tox.ini file
+
+# The scenario tests are the current set of smoke tests. This is sufficient
+# for now, but we will be able to amend this set as needed.
+tox -e scenariov2 -c f5.tox.ini --sitepackages -- \
+  --meta ${EXCLUDE_DIR}/${EXCLUDE_FILE} \
+  -lvv --tb=short \
```
Comment (zancas) [inline systest/scripts/run_smoke_tests.sh]: Why short?

**Comment by pjbreaux** [comment]
retest this please

**Comment by pjbreaux** [comment]
@ssorenso: Okay, I believe the changes are complete. We are not getting results uploaded to trtl because there are differences between my Jenkins server and the production server. That labels the jobs differently, so the results don't go to the correct spot when they are run from my Jenkins server. That's ok, as long as they do the right thing in nightly. We can merge in liberty and I'll check the results in the morning. Thanks!

**Comment by pjbreaux** [comment]
@zancas: I forgot you were the assignee. If you get this, we are ready to merge.

**Pull request merged**